hls4ml/templates/vivado/nnet_utils/nnet_sepconv_stream.h

```diff
@@ -23,7 +23,7 @@ void depthwise_product(data_T data[CONFIG_T::kernel_size * CONFIG_T::n_chan], re
 
     #pragma HLS ARRAY_PARTITION variable=mult complete
 
-    #pragma HLS ALLOCATION operation instances=mul limit=CONFIG_T::multiplier_limit
+    #pragma HLS ALLOCATION operation instances=mul limit=CONFIG_T::mult_config::multiplier_limit
```

Comment:
This technically works, but we deliberately didn't use the `mult_config` in `depthwise_product`. In 2D the config has the `multiplier_limit` and we should use the same approach. So below [this ](https://github.com/fastmachinelearning/hls4ml/blob/main/hls4ml/backends/vivado/passes/convolution_templates.py#L43) line we should add to [these](https://github.com/fastmachinelearning/hls4ml/blob/main/hls4ml/backends/vivado/passes/convolution_templates.py#L146-L147) lines.

Reply:
Ok, I was suspecting that there was something else to it, but not familiar with the code base yet. I'll start by adding a pytest that trig the issue, add the fix you suggest and force push to this PR.